src/config.rs

```diff
@@ -0,0 +1,86 @@
+use ansi_term::Color::Red;
+use ansi_term::Style;
+use atty::Stream::Stdout;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Config {
+	#[structopt(
+		parse(from_str = "parse_color"),
+		default_value = "auto",
+		raw(possible_values = "&[\"never\", \"auto\", \"always\"]"),
+		long = "color"
+	)]
+	pub color: Colors,
+	#[structopt(subcommand)]
+	pub action: Action,
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Action {
```

Comment:
following `clap` terminology, we can call it `Command` or `Subcommand`.

Reply:
Though I'm not sure why I'm writing this, "action" is probably nice as well. Nevermind.